From d6c844dc54a7532c15b233a5d5d392b3573c3f5e Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Wed, 6 Mar 2024 18:43:00 -0800
Subject: [PATCH] ==== DAMOS filter type YOUNG ====

Subject: [PATCH] mm/damon: add a DAMOS filter type for page granularity access recheck

Changes from RFC v2
(https://lore.kernel.org/r/20240311204545.47097-1-sj@kernel.org)
- Add documentation
- Add Tested-by: Honggyu Kim <honggyu.kim@sk.com>

Changes from RFC v1
(https://lore.kernel.org/r/20240307030013.47041-1-sj@kernel.org)
- Mark the folio as old if it was young
- Rename __damon_pa_young() to damon_folio_young_one()

DAMON provides its best-effort accuracy-overhead tradeoff under the
user-defined ranges of acceptable level of the monitoring accuracy and
overhead.  A recent discussion for tiered memory management support from
DAMON[1] revealed that the best-effort may not suffice in some use
cases.  The discussion further concluded finding memory regions of
specific access pattern with low via DAMON first, and then double
checking the access of the region again in a finer granularity could
help increasing the accuracy while keeping the low overhead.

Add a new type of DAMOS filter, namely 'young' for such a case.  The
type of filter is applied to each page of the memory region of DAMOS
target memory region, and check if the page is accessed since the last
check.  Then the page is filtered out or in if 'matching' parameter is
'true' or 'false', respectively.

Because this is a filter type that applied in page granularity, the
support depends on DAMON operations set, similar to 'anon' and 'memcg'
DAMOS filter types.  Implement the support on the DAMON operations set
for the physical address space, 'paddr', since the expected usage is
based on the physical address space.

[1] https://lore.kernel.org/r/20240227235121.153277-1-sj@kernel.org

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/0tpBKqMR | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/0tpBKqMR

diff --git a/damon_meta_changes/0tpBKqMR b/damon_meta_changes/0tpBKqMR
new file mode 100644
index 0000000000000..e69de29bb2d1d
-- 
2.39.2

